nibabel/analyze.py

```diff
@@ -990,6 +1002,17 @@ def from_filename(klass, filename, mmap=True):
             `mmap` value of True gives the same behavior as ``mmap='c'``.  If
             image data file cannot be memory-mapped, ignore `mmap` value and
             read array from file.
+        keep_file_open : { 'auto', True, False }, optional, keyword only
+            `keep_file_open` controls whether a new file handle is created
+            every time the image is accessed, or a single file handle is
+            created and used for the lifetime of this ``ArrayProxy``. If
+            ``True``, a single file handle is created and used. If ``False``,
+            a new file handle is created every time the image is accessed. If
+            ``'auto'`` (the default), and the optional ``indexed_gzip``
+            dependency is present, a single file handle is created and
+            persisted. If ``indexed_gzip`` is not available, behaviour is the
+            same as if ``keep_file_open is False``. If ``file_like`` is an
+            open file handle, this setting has no effect.
```

Comment:
Aah, I was just copying/pasting that comment block. 

I was wondering about this sort of thing. In my own projects, I tend to define and document arguments on the function/methods/classes that they are used in, and then pass them through with `**kwargs` for functions/methods/classes which call them. 

But for this PR, I was following what seems to be the convention in `nibabel`, where arguments are explicitly defined at each level. What is your take on this?

Reply:
Not sure quite what you mean by duplication. (Maybe a concrete example of the kind of change you're proposing would be useful?) But on the whole, I think this is looking good as is. I'll have another read through now.

Reply:
By duplication, I mean that the `mmap=True, keep_file_open='auto'` arguments, and associated docstrings, are defined multiple times, but only really need to be defined in the `ArrayProxy`, and in `analyze.py`, `spm99analyze.py` (and future classes) could be passed through with `**kwargs` (and have docstrings saying "See ArrayProxy for details on the other keyword arguments). 

I just like to remove as much duplication as possible. But in this case it's a minor point.

Reply:
I see. I think the idea is to keep docstrings as informative as possible. It's brittle, but I think that's good because API changes shouldn't be easy to do accidentally. And it's harder for your docs to get out of sync with your behavior.